src/compiler/preprocessor.rs

```diff
@@ -67,27 +148,63 @@ fn process_pp_form(
         Ok(())
     };
 
-    let included: Option<IncludeDesc> = body
+    let include_type: Option<IncludeType> = body
         .proper_list()
         .map(|x| x.iter().map(|elt| elt.atomize()).collect())
         .map(|x: Vec<SExp>| {
             match &x[..] {
                 [SExp::Atom(kw, inc), SExp::Atom(nl, fname)] => {
-                    if "include".as_bytes().to_vec() == *inc {
-                        return Ok(Some(IncludeDesc {
+                    if inc == b"include" {
+                        return Ok(Some(IncludeType::Basic(IncludeDesc {
                             kw: kw.clone(),
                             nl: nl.clone(),
+                            kind: None,
                             name: fname.clone(),
-                        }));
+                        })));
                     }
                 }
-                [SExp::Atom(kw, inc), SExp::QuotedString(nl, _, fname)] => {
-                    if "include".as_bytes().to_vec() == *inc {
-                        return Ok(Some(IncludeDesc {
-                            kw: kw.clone(),
-                            nl: nl.clone(),
-                            name: fname.clone(),
-                        }));
+
+                [SExp::Atom(kl, embed_file), SExp::Atom(_, name), SExp::Atom(_, kind), SExp::Atom(nl, fname)] => {
+                    if embed_file == b"embed-file" {
```

Comment:
TODO: Add the syntax rules for `embed-file`:
must exactly one of "hex", "bin", "sexp" be specified?